collections/indexes/multi.go

```diff
@@ -121,6 +123,40 @@ func (m *Multi[ReferenceKey, PrimaryKey, Value]) MatchExact(ctx context.Context,
 	return m.Iterate(ctx, collections.NewPrefixedPairRange[ReferenceKey, PrimaryKey](refKey))
 }
 
+// RefKeys returns a list of all the MultiIterator's reference keys (may contain duplicates).
+// Enable the "unique" argument to get a unique list of reference keys (the reference key must be comparable)
+func (m *Multi[ReferenceKey, PrimaryKey, Value]) RefKeys(ctx context.Context, unique bool) ([]ReferenceKey, error) {
+	// sanity check - enabled unique with non-comparable ReferenceKey type
+	if unique && !reflect.ValueOf((*ReferenceKey)(nil)).Comparable() {
+		return nil, fmt.Errorf("cannot retrieve unique reference keys since type is not comparable: %T", reflect.TypeOf((*ReferenceKey)(nil)))
+	}
+
+	iter, err := m.refKeys.Iterate(ctx, nil)
```

Comment:
i think we should RawIterate getting bytes, then use the bytes as map index. (can be efficiently done with unsafe)
then convert them to the real type later using the key codec

Reply:
not sure I follow. `IterateRaw` uses an untyped range but returns a typed iterator. So we're not getting bytes out of it. Can you write a small pseudo code of what you meant? (or explain in other words?)

Reply:
@julienrbrt any chance you get what he meant?

Reply:
Not really. @testinginprod care to ellaborate?

Reply:
@testinginprod trying again...
The work is almost done, I would really like see this code merged, and it's stuck for some time (@julienrbrt)

Reply:
Sorry for being slow!

So:
1. we cannot use reflection.
2. we should iterate the map using the raw bytes and comparing the raw bytes for equality

Reply:
Also maybe to fast track merging, we could split the PR in two,and push the multi refkeys method to another PR since it is more contentious than the triple reverse range

Reply:
now comparing bytes for equality and not using reflect :)
Hope it's ok, see here: https://github.com/oren-lava/cosmos-sdk/commit/a56e94317477e96b81982f69e82989ea91adb699